test/Quest1155.t.sol

```diff
@@ -0,0 +1,306 @@
+import {Events} from "./helpers/Events.sol";
+
+contract TestQuest1155 is Test, Errors, Events {
+    using LibClone for address;
+    using LibString for uint256;
+
+    Quest1155 quest;
+    address sampleERC1155;
+    uint256 END_TIME = 1_000_000_000;
+    uint256 START_TIME = 1_000_000;
+    uint256 TOTAL_PARTICIPANTS = 300;
+    uint256 TOKEN_ID = 1;
+    uint16 QUEST_FEE = 2000; // 20%
+    address protocolFeeRecipient = makeAddr("protocolFeeRecipient");
+    address questFactoryMock;
+    address participant = makeAddr(("participant"));
+    address owner = makeAddr(("owner"));
+
+    function setUp() public {
+        questFactoryMock = address(new QuestFactoryMock());
+        sampleERC1155 = address(new SampleERC1155());
+        address payable questAddress = payable(address(new Quest1155()).cloneDeterministic(keccak256(abi.encodePacked(msg.sender, "SALT"))));
+        quest = Quest1155(questAddress);
+        vm.prank(questFactoryMock);
+
+        quest.initialize(
+            sampleERC1155,
+            END_TIME,
+            START_TIME,
+            TOTAL_PARTICIPANTS,
+            TOKEN_ID,
+            QUEST_FEE,
+            protocolFeeRecipient
+        );
+
+        vm.warp(START_TIME + 1);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                              INITIALIZE
+    //////////////////////////////////////////////////////////////*/
+    function test_initialize() public {
+        assertEq(sampleERC1155, quest.rewardToken(), "rewardTokenAddress not set");
+        assertEq(END_TIME, quest.endTime(), "endTime not set");
+        assertEq(START_TIME, quest.startTime(), "startTime not set");
+        assertEq(TOTAL_PARTICIPANTS, quest.totalParticipants(), "totalParticipants not set");
+        assertEq(TOKEN_ID, quest.tokenId(), "tokenId not set");
+        assertEq(QUEST_FEE, quest.questFee(), "questFee not set");
+        assertEq(protocolFeeRecipient, quest.protocolFeeRecipient(), "protocolFeeRecipient not set");
+        assertEq(questFactoryMock, quest.owner(), "owner should be set");
+    }
+
+    function test_RevertIf_initialize_EndTimeInPast() public {
+        vm.warp(END_TIME + 1);
+        questFactoryMock = address(new QuestFactoryMock());
+        sampleERC1155 = address(new SampleERC1155());
+        address payable questAddress = payable(address(new Quest1155()).cloneDeterministic(keccak256(abi.encodePacked(msg.sender, "SALT"))));
+        quest = Quest1155(questAddress);
+        vm.prank(questFactoryMock);
+
+        vm.expectRevert(abi.encodeWithSelector(EndTimeInPast.selector));
+        quest.initialize(
+            sampleERC1155,
+            END_TIME,
+            START_TIME,
+            TOTAL_PARTICIPANTS,
+            TOKEN_ID,
+            QUEST_FEE,
+            protocolFeeRecipient
+        );
+    }
+
+    function test_RevertIf_initialize_EndTimeLessThanOrEqualToStartTime() public {
+        vm.warp(START_TIME - 1); // reset time
```

Comment:
I don't think this is needed - I believe it'll reset to the startup state for each test. Not positive though.